run/image-processing/app.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 Google LLC. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const app = express();
+
+app.use(bodyParser.json());
+
+// [START run_imageproc_controller]
+const image = require('./image');
+
+app.post('/', async (req, res) => {
+  if (!req.body) {
+    const msg = 'no Pub/Sub message received';
+    console.error(`error: ${msg}`);
+    res.status(400).send(`Bad Request: ${msg}`);
+    return;
+  }
+  if (!req.body.message || !req.body.message.data) {
+    const msg = 'invalid Pub/Sub message format';
+    console.error(`error: ${msg}`);
+    res.status(400).send(`Bad Request: ${msg}`);
+    return;
+  }
+
+  // Decode the Pub/Sub message.
+  const pubSubMessage = req.body.message;
+  pubSubMessage.data = Buffer.from(pubSubMessage.data, 'base64')
+    .toString()
+    .trim();
+  let data;
+  try {
+    data = JSON.parse(pubSubMessage.data);
```

Comment:
I agree with @fhinkel I would just decode and parse in this same try, with the error:

```
Invalid Pub/Sub message: data is not valid base64 encoded JSON
``